cnf-tests/testsuites/e2esuite/ovs_qos/ovs_qos.go

```diff
@@ -0,0 +1,980 @@
+	if hostNetwork {
+		pod = pods.DefinePodOnHostNetwork(namespaces.OVSQOSTest, nodeName)
+	} else {
+		pod = pods.DefinePodOnNode(namespaces.OVSQOSTest, nodeName)
+	}
+	pod = pods.RedefineWithCommand(pod, []string{"/bin/bash", "-c"}, []string{"iperf3 -s -J -1;"})
+	pod = pods.RedefineWithRestartPolicy(pod, corev1.RestartPolicyNever)
+	return pod
+}
+
+func defineSenderPod(nodeName string, hostNetwork bool, receiverIP string) *corev1.Pod {
+	var pod *corev1.Pod
+	if hostNetwork {
+		pod = pods.DefinePodOnHostNetwork(namespaces.OVSQOSTest, nodeName)
+	} else {
+		pod = pods.DefinePodOnNode(namespaces.OVSQOSTest, nodeName)
+	}
+	pod = pods.RedefineWithCommand(pod, []string{"/bin/bash", "-c"}, []string{fmt.Sprintf("iperf3 -u -b 0 -c %s -t 10 -J;", receiverIP)})
+	pod = pods.RedefineWithRestartPolicy(pod, corev1.RestartPolicyNever)
+	return pod
+}
+
+func findQoSRateWithNodeSelector(qosType func(string) float64, qosNodeSelector string) (string, float64, error) {
+	mcList, err := client.Client.MachineConfigs().List(context.Background(), metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mc := range mcList.Items {
+		enables, err := isMCLimiting(mc, qosType)
+		if err != nil {
+			return "", -1, err
+		}
+		if enables != -1 {
+			mcLabel, found := mc.ObjectMeta.Labels["machineconfiguration.openshift.io/role"]
+			if !found {
+				continue
+			}
+			nodeSelector, err := findQoSNodeSelectorByMCLabel(mcLabel)
+			if err != nil {
+				continue
+			}
+			if qosNodeSelector != "" && qosNodeSelector != nodeSelector {
+				continue
+			}
+
+			return nodeSelector, enables, nil
+		}
+	}
+
+	return "", -1, fmt.Errorf("cannot find a machine configuration with specified qosNodeSelector that applies limitation")
+}
+
+func isMCLimiting(mc mcfgv1.MachineConfig, qosType func(string) float64) (float64, error) {
+	ignitionConfig := igntypes.Config{}
+	if mc.Spec.Config.Raw == nil {
+		return -1, nil
+	}
+
+	err := json.Unmarshal(mc.Spec.Config.Raw, &ignitionConfig)
+	if err != nil {
+		return -1, fmt.Errorf("failed to unmarshal ignition config %v", err)
+	}
+
+	for _, unit := range ignitionConfig.Systemd.Units {
+		bitLimit := qosType(unit.Contents)
+		if bitLimit != -1 {
+			return bitLimit, nil
+		}
+
+	}
+	return -1, nil
+}
+
+func isEgressLimiting(s string) float64 {
+	egressRegexp := regexp.MustCompile("create qos .* other-config:max-rate=([0-9]*)")
+	m := egressRegexp.FindStringSubmatch(s)
+	if len(m) < 2 {
+		return -1
+	}
+	bitRate, err := strconv.ParseFloat(m[1], 64)
+	if err != nil {
+		return -1
+	}
+	return bitRate
+}
+
+func isIngressLimiting(s string) float64 {
+	ingressRegexp := regexp.MustCompile("set interface .* ingress_policing_rate=([0-9]*)")
+	m := ingressRegexp.FindStringSubmatch(s)
+	if len(m) < 2 {
+		return -1
+	}
+	bitRate, err := strconv.ParseFloat(m[1], 64)
+	if err != nil {
+		return -1
+	}
+	// ingress QoS is set in kbits
+	return bitRate * 1000
+}
+
+func findQoSNodeSelectorByMCLabel(mcLabel string) (string, error) {
```

Comment:
we have this already multiple times in the code

please lets make this https://github.com/openshift-kni/cnf-features-deploy/blob/master/cnf-tests/testsuites/validationsuite/cluster/validation.go#L416

to be under the pkg/machineconfig so all the tests can reuse it.

follow up PR do the same change for x32 and sctp tests